src/recorder/text.rs

```diff
@@ -38,3 +38,23 @@ pub fn halfvec_out(vector: VectBorrowed<'_, f16>) -> String {
     result.push(']');
     result
 }
+
+pub fn validate(text: String) -> bool {
+    let text = text.trim();
+    if !text.starts_with('[') || !text.ends_with(']') {
+        return false;
+    }
+    let elements: Vec<&str> = text[1..text.len() - 1]
+        .split(',')
+        .map(|s| s.trim())
+        .collect();
+    if elements.is_empty() {
+        return false;
+    }
+    for element in elements {
+        if element.is_empty() || element.parse::<f32>().is_err() {
+            return false;
+        }
+    }
+    true
+}
```

Comment:
I think we can just guide users to use `quote_literal` and `quote_ident` in SQL.